packages/hash/api/src/model/page.model.ts

```diff
@@ -301,6 +307,14 @@ class __Page extends Entity {
     /** @todo: remove when modifying links no longer creates new versions of the source entity */
     await this.refetchLatestVersion(client);
   }
+
+  toGQLPageEntity(): UnresolvedGQLPage {
+    return {
+      ...this.toGQLEntity(),
+      parentPageId: this.parentEntityId,
```

Comment:
**Blocking:** I think we should add a `parentPage` field resolver instead. That'll let the client access things like `parentPage.properties.title`, or whatever they wish.

Reply:
I do agree with having a `pageParent` resolver and letting it be a part of a `page`, but I also think that the `parentPageId` is relevant for not fetching the the entire `parentPage` when we just need a reference. How are we tackling this sort of setup currently? I assume we have something like this with the `EntityTypeType`?

Reply:
> I also think that the parentPageId is relevant for not fetching the the entire parentPage when we just need a reference
This is going to be addressed once we have a generalised caching solution in place. I don't think it's a good idea for us to introduce too many special cased performance boosts now, that we will end up just having to remove down the line anyway.

Reply:
> Follow up question, would the parentPage field resolver also fetch the outgoing link of the page?

Yes

Reply:
Good point, I didn't consider this being a caching issue, but rather a N+1 sort of deal. Since GraphQL is restrictive in that sense, though, I think my concerns are not based on a concrete issue :)

Reply:
> How are we tackling this sort of setup currently? I assume we have something like this with the EntityTypeType?

I'm not sure I fully follow this question, if you're asking what the current approach for custom field resolvers is an example would be the existing `contents` field resolver for the `Page` GQL type.

Reply:
We implemented `parentPage` and `parentPageEntityId` field resolvers